internal/validators/os/os_validator.go

```diff
@@ -0,0 +1,108 @@
+// Package os handles OS image rule reconciliation.
+package os
+
+import (
+	"fmt"
+
+	"github.com/canonical/gomaasclient/api"
+	"github.com/canonical/gomaasclient/entity"
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/go-logr/logr"
+
+	"github.com/validator-labs/validator-plugin-maas/api/v1alpha1"
+	"github.com/validator-labs/validator-plugin-maas/internal/constants"
+	vapi "github.com/validator-labs/validator/api/v1alpha1"
+	"github.com/validator-labs/validator/pkg/types"
+	"github.com/validator-labs/validator/pkg/util"
+)
+
+// ImageRulesService is a service for reconciling OS image rules
+type ImageRulesService struct {
+	log logr.Logger
+	api api.BootResources
+}
+
+// NewImageRulesService returns a ImageRulesService
+func NewImageRulesService(log logr.Logger, api api.BootResources) *ImageRulesService {
+	return &ImageRulesService{
+		log: log,
+		api: api,
+	}
+}
+
+// ReconcileMaasInstanceImageRule reconciles a MAAS instance image rule from the MaasValidator config
+func (s *ImageRulesService) ReconcileMaasInstanceImageRule(rule v1alpha1.ImageRule) (*types.ValidationRuleResult, error) {
+
+	vr := buildValidationResult(rule)
+
+	errs, details := s.findBootResources(rule)
+
+	s.updateResult(vr, errs, constants.ErrMsg, details...)
+
+	if len(errs) > 0 {
+		return vr, errs[0]
+	}
+	return vr, nil
+}
+
+// buildValidationResult builds a default ValidationResult for a given validation type
+func buildValidationResult(rule v1alpha1.ImageRule) *types.ValidationRuleResult {
+	state := vapi.ValidationSucceeded
+	latestCondition := vapi.DefaultValidationCondition()
+	latestCondition.Details = make([]string, 0)
+	latestCondition.Failures = make([]string, 0)
+	latestCondition.Message = fmt.Sprintf("All %s checks passed", constants.MaasInstance)
+	latestCondition.ValidationRule = rule.Name
+	latestCondition.ValidationType = constants.MaasInstance
```

Comment:
Please add meaningful constants for the MAAS rule types, similar to https://github.com/validator-labs/validator-plugin-aws/blob/main/internal/constants/constants.go#L9. Like `maas-image`, `maas-internal-dns`, etc.